### ISSUE TITLE
Support PSALM function type signatures

### ISSUE THREAD

**Issue by klesun** [klesun/deep-assoc-completion#163]
Like this one:
```php
/**
 * @template T
 */
class LazyValue
{
    /**
     * @psalm-var Closure(): T
     */
    private Closure $callable;

```
There were docs on their page somewhere...

**Comment by andrew-demb** [comment]
Here it is :)
https://psalm.dev/docs/annotating_code/type_syntax/callable_types/

**Comment by zoltanka** [comment]
Well yeah plus one to this. The combination of psalm and this would be an extremely powerful development tool.

**Comment by klesun** [comment]
A bit more discussion about PSALM function types support: https://github.com/klesun/deep-assoc-completion/issues/137
So far I see this does not work, will work on it:
![image](https://user-images.githubusercontent.com/5202330/102725402-26447580-431f-11eb-9471-bba9e6b9a6a7.png)

**Comment by klesun** [comment]
Released in [2020.12.21.001](https://plugins.jetbrains.com/plugin/9927-deep-assoc-completion/versions/stable/105995)
![Uploading image.png…]()
